src/app/api/product-reviews/route.ts

```diff
@@ -15,53 +16,60 @@ const requestType = z.object({
 });
 
 export async function POST(request: Request) {
-  let body = undefined;
-  try {
-    body = await request.json();
-  } catch (err) {
-    return response400("Incorect format");
-  }
-  const results = requestType.safeParse(body);
-  if (!results.success!) {
-    return response400("Incorect format");
-  }
-  const parsedBody = results.data;
-
-  try {
-    let product = await getProduct(parsedBody.product_id);
-    if (product === undefined) {
-      return response400("There is no product with this ID");
+  const user = await getUserServerSide();
+  if (user === undefined) {
+    console.log("User not logged in");
+  } else {
+    let body = undefined;
+    try {
+      body = await request.json();
+    } catch (err) {
+      return response400("Incorect format");
     }
-    const parsedBodyWithUserDetails: z.infer<typeof requestType> & {
-      user_id: string;
-      product_name: string;
-    } = {
-      ...parsedBody,
-      // IMPORTANT: this is hardcoded because we don't yet have user management.
-      // TO BE DELETED once we integrate authentication
-      user_id: "Diana13",
-      product_name: "test",
-    };
+    const results = requestType.safeParse(body);
+    if (!results.success!) {
+      return response400("Incorect format");
+    }
+    const parsedBody = results.data;
+
+    try {
+      let product = await getProduct(parsedBody.product_id);
+      if (product === undefined) {
+        return response400("There is no product with this ID");
+      }
+      const parsedBodyWithUserDetails: z.infer<typeof requestType> & {
```

Comment:
I don't think we need this anymore. This was a hack that we needed because we didn't have the user.

You already have all the info you need into either `user` or `parsedBody` variables.